File: tests/human/x/php/group_by_having.php
<?php
$people = [
    ["name" => "Alice", "city" => "Paris"],
    ["name" => "Bob", "city" => "Hanoi"],
    ["name" => "Charlie", "city" => "Paris"],
    ["name" => "Diana", "city" => "Hanoi"],
    ["name" => "Eve", "city" => "Paris"],
    ["name" => "Frank", "city" => "Hanoi"],
    ["name" => "George", "city" => "Paris"],
];
$groups = [];
foreach ($people as $p) {
    $groups[$p['city']][] = $p;
}
$big = [];
foreach ($groups as $city => $persons) {
    if (count($persons) >= 4) {
        $big[] = ["city" => $city, "num" => count($persons)];
    }
}
_print($big);

function _print(...$args) {
    $parts = [];
    foreach ($args as $a) {
        if (is_array($a) || is_object($a)) {
            $parts[] = json_encode($a);
        } else {
            $parts[] = strval($a);
        }
    }
    echo implode(' ', $parts), PHP_EOL;
}
?>

File: tests/human/rust/group_by_having.rs
use std::collections::BTreeMap;

#[derive(Clone)]
struct Person { name: &'static str, city: &'static str }

fn main() {
    let people = vec![
        Person { name: "Alice", city: "Paris" },
        Person { name: "Bob", city: "Hanoi" },
        Person { name: "Charlie", city: "Paris" },
        Person { name: "Diana", city: "Hanoi" },
        Person { name: "Eve", city: "Paris" },
        Person { name: "Frank", city: "Hanoi" },
        Person { name: "George", city: "Paris" },
    ];

    let mut groups: BTreeMap<&str, Vec<&Person>> = BTreeMap::new();
    for p in &people {
        groups.entry(p.city).or_default().push(p);
    }

    let mut big = Vec::new();
    for (city, group) in &groups {
        if group.len() >= 4 {
            big.push((city, group.len()));
        }
    }

    let json_output = serde_json::to_string(&big.iter().map(|(city, num)| serde_json::json!({"city": city, "num": num})).collect::<Vec<_>>()).unwrap();
    println!("{}", json_output);
}
